src/components/TodoItem.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export class TodoItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false,
+    };
+    this.editingInput = React.createRef();
+  }
+
+  componentDidUpdate() {
+    if (this.state.isEditing) {
```

Comment:
log something inside this condition and try to edit, count of logs will surprise you

Will be better to focus input in the place where you enable editing